locked-nft/contracts/NFTLocker.cdc

```diff
@@ -0,0 +1,180 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract NFTLocker {
+
+    pub event ContractInitialized()
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+    pub event NFTLocked(
+        id: UInt64,
+        to: Address?,
+        lockedAt: UInt64,
+        lockedUntil: UInt64,
+        duration: UInt64,
+        nftType: Type
+    )
+    pub event NFTUnlocked(
+        id: UInt64,
+        from: Address?,
+        nftType: Type
+    )
+
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+
+    pub var totalLockedTokens:      UInt64
+    access(self) let lockedTokens:  {Type: {UInt64: LockedData}}
+
+    pub struct LockedData {
+        pub let id: UInt64
+        pub let owner: Address
+        pub let lockedAt: UInt64
+        pub let lockedUntil: UInt64
+        pub let duration: UInt64
+        pub let nftType: Type
+
+        init (id: UInt64, owner: Address, duration: UInt64, nftType: Type) {
+            let key = NFTLocker.getLockedTokenKey(id: id, nftType: nftType)
```

Comment:
We don't need the key concept anymore as we start using the double dictionary

Reply:
Removed. Thanks.